src/middlewared/middlewared/plugins/disk_/swap_configure.py

```diff
@@ -67,11 +68,17 @@ class DiskService(Service):
             lambda d: d['partition_type'] in valid_swap_part_uuids and d['name'] not in used_partitions_in_mirror,
             all_partitions.values()
         ):
-            if osc.IS_FREEBSD and not any(swap_part[k] in existing_swap_devices for k in ('encrypted_provider', 'path')):
+            if osc.IS_FREEBSD and not any(
+                swap_part[k] in existing_swap_devices for k in ('encrypted_provider', 'path')
+            ):
                 # Try to save a core dump from that.
                 # Only try savecore if the partition is not already in use
                 # to avoid errors in the console (#27516)
-                await run('savecore', '-z', '-m', '5', '/data/crash/', f'/dev/{swap_part["name"]}', check=False)
+                cp = await run('savecore', '-z', '-m', '5', '/data/crash/', f'/dev/{swap_part["name"]}', check=False)
+                if cp.returncode:
+                    self.middleware.logger.error(
+                        'Failed to savecore for "%s": ', f'/dev/{swap_part["name"]}', cp.stderr.decode()
+                    )
```

Comment:
Have you tested this to make sure savecore doesnt log errors even when there has not been crashdumps and it tried to run savecore? It has been a problem before, reason we didnt log this.

Reply:
From what I have seen, that happens only if the partition is in use where we get `Operation not permitted` errors. We are handling that bit above by checking that it does not come in existing swap devices